File: src/test/java/ResumeTests.java
import com.fasterxml.jackson.core.JsonProcessingException;
import lombok.extern.slf4j.Slf4j;
import main.utils.resume.GuildResumeCache;
import main.utils.resume.ResumableTrack;
import main.utils.resume.ResumeData;
import org.junit.jupiter.api.Test;

import java.util.stream.IntStream;

import static org.springframework.test.util.AssertionErrors.assertEquals;
import static org.springframework.test.util.AssertionErrors.assertNotNull;

@Slf4j
public class ResumeTests {

    @Test
    void testResumableDataToJSON() {
        final var tracks = UtilsTest.getManyTracksWithRequester(10);
        final var resumableTracks = tracks.stream()
                .map(ResumableTrack::new)
                .toList();
        final var resumeData = new ResumeData("vcId", "acId", resumableTracks);
        final var json = resumeData.toString();
        log.info("JSON: {}", json);
        assertNotNull("Checking if JSON is valid", json);
    }

    @Test
    void testResumableDataJSONToObject() {
        final var tracks = UtilsTest.getManyTracksWithRequester(10);
        final var resumableTracks = ResumableTrack.audioTracksToResumableTracks(tracks);
        final var resumeData = new ResumeData("vcId", "acId", resumableTracks);
        final var json = resumeData.toString();

        try {
            final var revertedData = ResumeData.fromJSON(json);

            assertNotNull("Checking if data is null", revertedData);
            assertNotNull("Checking if voice channel ID is null", revertedData.getChannel_id());
            assertEquals("Checking track list length", 10, revertedData.getTracks().size());


            for (int i = 0; i < revertedData.getTracks().size(); i++) {
                assertNotNull("Checking if track %d is not null".formatted(i), revertedData.getTracks().get(i));
            }
        } catch (JsonProcessingException e) {
            log.error("Couldn't load tracks for a test!", e);
        }
    }

    @Test
    void testSingularResumeCacheLoadAndSave() {
        final var resumeCache = new GuildResumeCache("1");
        final var tracks = UtilsTest.getManyTracksWithRequester(10);
        final var resumableTracks = ResumableTrack.audioTracksToResumableTracks(tracks);
        final var resumeData = new ResumeData("vcId", "acId", resumableTracks);

        try {
            resumeCache.setTracks(resumeData);
            final var loadedData = resumeCache.loadData();

            assertEquals("Testing resume cache", resumeData.toString(), loadedData.toString());
        } catch (JsonProcessingException e) {
            log.error("Couldn't load tracks for a test!", e);
        }

    }

    @Test
    void testManyResumeCacheLoadAndSave() {
        final var caches = IntStream.range(0, 10)
                .mapToObj(i -> new GuildResumeCache(String.valueOf(i)));
        final var tracks = UtilsTest.getManyTracksWithRequester(50);
        final var resumableTracks = ResumableTrack.audioTracksToResumableTracks(tracks);
        final var resumeData = new ResumeData("vcID", "acId", resumableTracks);

        caches.forEach(cache -> {
            cache.setTracks(resumeData);
            final ResumeData loadedData;
            try {
                loadedData = cache.loadData();
                assertEquals("Testing resume cache for cache with ID " + cache.getCacheID(), resumeData.toString(), loadedData.toString());

            } catch (JsonProcessingException e) {
                log.error("Couldn't load tracks for a test!", e);
            }
        });
    }
}

File: src/test/kotlin/ResumeTests.kt
//import com.fasterxml.jackson.core.JsonProcessingException
//import dev.minn.jda.ktx.util.SLF4J
//import main.audiohandlers.models.Requester
//import main.utils.resume.GuildResumeCache
//import main.utils.resume.ResumableTrack
//import main.utils.resume.ResumableTrack.Companion.toResumableTracks
//import main.utils.resume.ResumeData
//import kotlin.test.Test
//import kotlin.test.assertEquals
//import kotlin.test.assertNotNull
//
//class ResumeTests {
//
//    val logger by SLF4J
//
//    @Test
//    fun testResumableDataToJson() {
//        val tracks = mockAudioTracks(10)
//        val resumableTracks = tracks.map { ResumableTrack(it, Requester("userid", it.identifier)) }
//        val resumeData = ResumeData("vcId", resumableTracks)
//        val json = resumeData.toString()
//        logger.info("Json: $json")
//        assertNotNull(json, "Checking if Json is valid.")
//    }
//
//    @Test
//    fun testResumeDataJsonToObject() {
//        val tracks = mockAudioTracksWithRequester(10)
//        val resumableTracks = tracks.toResumableTracks()
//        val resumeData = ResumeData("vcId", resumableTracks)
//        val json = resumeData.toString()
//
//        try {
//            val revertedData = ResumeData.fromJSON(json)
//
//            assertNotNull(revertedData, "Checking if data is null")
//            assertNotNull(revertedData.channel_id, "Checking if voice channel ID is null")
//            assertEquals(10, revertedData.tracks.size, "Checking track list length")
//
//            revertedData.tracks.forEach { track ->
//                assertNotNull(track, "Checking if track ${track.info.identifier} is null")
//            }
//
//        } catch (e: JsonProcessingException) {
//            logger.error("Couldn't load tracks for a test!", e)
//        }
//    }
//
//    @Test
//    fun testSingularResumeCacheLoadAndSave() {
//        val resumeCache = GuildResumeCache("1")
//        val tracks = mockAudioTracksWithRequester(10)
//        val resumableTracks = tracks.toResumableTracks()
//        val resumeData = ResumeData("vcId", resumableTracks)
//
//        try {
//            resumeCache.data = resumeData
//            val loadedData = resumeCache.data
//
//            assertNotNull(loadedData, "Checking if the loaded data is null")
//            assertEquals(resumeData.toString(), loadedData.toString(), "Testing equivalence of the cache")
//        } catch (e: JsonProcessingException) {
//            logger.error("Couldn't load tracks for a test!", e)
//        }
//    }
//
//    @Test
//    fun testManyResumeCachesLoadAndSave() {
//        val resumeCaches = (0 until 10).map { GuildResumeCache(it.toString()) }
//        val tracks = mockAudioTracksWithRequester(500)
//        val resumableTracks = tracks.toResumableTracks()
//        val resumeData = ResumeData("vcId", resumableTracks)
//
//        resumeCaches.forEach { resumeCache ->
//            try {
//                resumeCache.data = resumeData
//                val loadedData = resumeCache.data
//
//                assertNotNull(loadedData, "Checking if the loaded data is null")
//                assertEquals(resumeData.toString(), loadedData.toString(), "Testing equivalence of the cache")
//            } catch (e: JsonProcessingException) {
//                logger.error("Couldn't load tracks for a test!", e)
//            }
//        }
//    }
//}